node/node.go

```diff
@@ -623,50 +622,50 @@ func (n *nodeImpl) OnStart(ctx context.Context) error {
 			n.logger.Error("failed to emit the statesync start event", "err", err)
 		}
 
-		// FIXME: We shouldn't allow state sync to silently error out without
-		// bubbling up the error and gracefully shutting down the rest of the node
-		go func() {
-			n.logger.Info("starting state sync")
-			state, err := n.stateSyncReactor.Sync(ctx)
-			if err != nil {
-				n.logger.Error("state sync failed; shutting down this node", "err", err)
-				// stop the node
-				if err := n.Stop(); err != nil {
-					n.logger.Error("failed to shut down node", "err", err)
-				}
-				return
+		// RUN STATE SYNC NOW:
+		//
+		// TODO: Eventually this should run as part of some
```

Comment:
Do we have an issue for this we could link to? Possibly #4644? Though that seems more general.

Reply:
I think that's basically the issue that we'd use to do this work in.